pkg/apis/core/types.go

```diff
@@ -2226,6 +2226,23 @@ type ResourceRequirements struct {
 	Claims []ResourceClaim
 }
 
+// StorageRequirements describes the storage resource requirements for a volume.
+type StorageRequirements struct {
+	// Limits describes the maximum amount of compute resources allowed.
+	// +optional
+	Limits ResourceList
+	// Requests describes the minimum amount of compute resources required.
+	// If Request is omitted for a container, it defaults to Limits if that is explicitly specified,
+	// otherwise to an implementation-defined value
+	// +optional
+	Requests ResourceList
+	// Claims mirrors the corresponding compute resource requirements
+	// field. It got added by accident, has no meaning here and gets ignored
+	// silently.
+	// +optional
+	Claims []ResourceClaim
```

Comment:
Mark it as deprecated? And throw a warning when user set it?

Reply:
Mark as deprecated how? A free-form comment?

Do Go linters recognize `// Deprecated: ` for fields? That's worth checking.

For the warning, that would be a server-side warning, right? I know those exist, but don't know how they are implemented. Do you have a pointer for me?

Reply:
> Do Go linters recognize // Deprecated: for fields?

No, staticcheck only detects usage of deprecated structs. It still makes sense to use the `Deprecated` comment because it makes the intent clearer.

> For the warning, that would be a server-side warning, right?

https://kubernetes.io/blog/2020/09/03/warnings/#future-possibilities mentions "warnings about fields" as a future extension. I've asked on #sig-api-machinery whether it is possible today.

Reply:
@pohly  a server-side warning example
https://github.com/kubernetes/kubernetes/blob/c984d53b31655924b87a57bfd4d8ff90aaeab9f8/pkg/registry/core/persistentvolume/strategy.go#L79-L82

Reply:
That was already wired up for `PersistentVolumeClaimSpec`, so all I had to do was add a new if check there - done.

Reply:
I actually think we should drop the field (with an appropriate tombstone comment recording the json field name and protobuf tag used)

Reply:
"Drop the field" as in "not have it in the struct at all" (except for the comments)?

So breaking client-side validation (= YAML file has it, kubectl complains about unknown field) would be acceptable?

If we can do that, then let's remove the field. It would be cleaner. I just didn't dare... :sweat_smile:

Reply:
it was added in 1.27 documented as alpha, right? I'd drop the field in 1.28 and backport a change to 1.27 to clear the field on write from PVCs as well

Reply:
New version pushed which no longer has the field, only a comment about it in `core/v1/types.go`.